engines/zypper_auth/lib/zypper_auth/engine.rb

```diff
@@ -20,15 +25,23 @@ def verify_instance(request, logger, system)
         instance_data
       )
 
-      is_valid = verification_provider.instance_valid?
+      verification_provider.instance_valid?
+      is_valid = true # log the error only, don't raise auth error (for now)
       Rails.cache.write(cache_key, is_valid, expires_in: 1.hour)
       is_valid
     rescue InstanceVerification::Exception => e
-      logger.info('Access to the repos denied:')
-      logger.info(e.message)
-      logger.info("System login: #{system.login}, IP: #{request.remote_ip}")
-      Rails.cache.write(cache_key, false, expires_in: 1.hour)
-      false
+      details = [ "System login: #{system.login}", "IP: #{request.remote_ip}" ]
+      details << "Instance ID: #{verification_provider.instance_id}" if verification_provider.instance_id
+      details << "Billing info: #{verification_provider.instance_billing_info}" if verification_provider.instance_billing_info
```

Comment:
Hmm, we want to log failures only,  which implies that we do not recognize the provided billing information or that it is null. If we expect that our verification plugin has the proper billing and marketplace codes then this would be a superfluous test and log line, or am I missing something?

Reply:
That's the handler for `InstanceVerification::Exception` exceptions (raised by the plugin if something's wrong with the metadata) -- so yes, those detailed messages are for failures only.

Reply:
My point is that in the failure case we know the billing code is non existent, why try to log "nil"

But I am OK with it I am just not certain we need to write "nil" to our logs.

Reply:
We are never logging `nil`, though (there's a postfix `if` condition there -- `... if verification_provider.instance_billing_info`).